components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -57,6 +61,30 @@ void AIChatTabHelper::PDFA11yInfoLoadObserver::AccessibilityEventReceived(
 
 AIChatTabHelper::PDFA11yInfoLoadObserver::~PDFA11yInfoLoadObserver() = default;
 
+// static
+void AIChatTabHelper::BindPageContentExtractorHost(
+    content::RenderFrameHost* rfh,
+    mojo::PendingAssociatedReceiver<mojom::PageContentExtractorHost> receiver) {
+  DCHECK(rfh);
+  if (!rfh->IsInPrimaryMainFrame()) {
+    DVLOG(4) << "Not binding extractor host to non-main frame";
+    return;
+  }
+  auto* sender = content::WebContents::FromRenderFrameHost(rfh);
```

Comment:
i know you love `auto` but i can't infer what the type of `sender` is here.

Reply:
Really? It's literally in rhs - `content::WebContents::From...`. That's why I've used `auto` here. It's unneccessary repetition in the same line.